consensus/istanbul/backend/announce.go

```diff
@@ -1347,16 +1323,20 @@ func (sb *Backend) handleEnodeCertificateMsg(_ consensus.Peer, payload []byte) e
 	}
 
 	// Send a valEnodesShare message to the proxy when it's the primary
-	if sb.IsProxiedValidator() && sb.IsValidating() {
-		sb.proxiedValidatorEngine.SendValEnodesShareMsgToAllProxies()
+	if sb.announceManager.config.IsProxiedValidator && sb.IsValidating() {
+		sb.announceManager.proxyContext.GetProxiedValidatorEngine().SendValEnodesShareMsgToAllProxies()
 	}
 
 	return nil
 }
 
 // SetEnodeCertificateMsgMap will verify the given enode certificate message map, then update it on this struct.
 func (sb *Backend) SetEnodeCertificateMsgMap(enodeCertMsgMap map[enode.ID]*istanbul.EnodeCertMsg) error {
-	logger := sb.logger.New("func", "SetEnodeCertificateMsgMap")
+	return sb.announceManager.SetEnodeCertificateMsgMap(enodeCertMsgMap)
```

Comment:
i suppose this method can be removed, as only the proxy calls this and it could call the announce manager directly

Reply:
Not as easily, no. Because the proxy calls this via an interface method, not through the Backend struct type directly.

After the whole migration to the announce manager is done, I will consider options like maybe adding a `GetAnnounceManager` to these interfaces and remove all these forwarding methods from it.